src/spikeinterface/core/analyzer_extension_core.py

```diff
@@ -520,6 +520,49 @@ def _select_extension_data(self, unit_ids):
 
         return new_data
 
+    def get_templates(self, unit_ids=None):
+        """
+        Return average templates for multiple units.
+
+        Parameters
+        ----------
+        unit_ids: list or None
+            Unit ids to retrieve waveforms for
+
+        Returns
+        -------
+        templates: np.array
+            The returned templates (num_units, num_samples, num_channels)
+        """
+
+        templates = self.data["average"]
+
+        if unit_ids is not None:
+            unit_indices = self.sorting_analyzer.sorting.ids_to_indices(unit_ids)
+            templates = templates[unit_indices, :, :]
+
+        return np.array(templates)
+
+    def get_unit_template(self, unit_id):
+        """
+        Return average template for a single unit.
+
+        Parameters
+        ----------
+        unit_id:
```

Comment:
is this an int or a string?

Reply:
Well it depends,

There is currently no defined type for `unit_id` in SpikeInterface

Reply:
Aren't they strings | int. Right?

Reply:
I believe they are, but who says that won't change in the future?

Reply:
You. Right now. Lock it in and take control.